src/Test/QuickCheck/Laws/Data/EuclideanRing.purs

```diff
@@ -1,22 +1,24 @@
-module Test.QuickCheck.Laws.Data.ModuloSemiring where
-
-import Prelude
-
-import Control.Monad.Eff.Console (log)
-
-import Type.Proxy (Proxy())
-
-import Test.QuickCheck (QC(), quickCheck')
-import Test.QuickCheck.Arbitrary (Arbitrary)
-
--- | - Remainder: ```a / b * b + (a `mod` b) = a```
-checkModuloSemiring :: forall eff a. (ModuloSemiring a, Arbitrary a, Eq a) => Proxy a -> QC eff Unit
-checkModuloSemiring _ = do
-
-  log "Checking 'Remainder' law for ModuloSemiring"
-  quickCheck' 1000 remainder
-
-  where
-
-  remainder :: a -> a -> Boolean
-  remainder a b = a / b * b + (a `mod` b) == a
+module Test.QuickCheck.Laws.Data.EuclideanRing where
+
+import Prelude
+
+import Control.Monad.Eff.Console (log)
+
+import Type.Proxy (Proxy())
+
+import Test.QuickCheck (QC(), quickCheck')
+import Test.QuickCheck.Arbitrary (class Arbitrary)
+
+-- | - Remainder: ```a / b * b + (a `mod` b) = a```
+checkEuclideanRing
```

Comment:
The laws here have changed actually:

- Integral domain: `a /= 0` and `b /= 0` implies `a * b /= 0`
- Multiplicative Euclidean function: ``a = (a / b) * b + (a `mod` b)`` where `degree a > 0` and `degree a <= degree (a * b)`

I can update this stuff though unless you want to?